contracts/shared/ForwarderXReceiver.sol

```diff
@@ -28,22 +31,47 @@ abstract contract ForwarderXReceiver {
         bytes calldata _callData
     ) external payable onlyConnext {
         // Decode calldata
-        (address fallbackAddress, bytes memory data) = abi
-            .decode(_callData, (address, bytes));
+        (address fallbackAddress, bytes memory data) = abi.decode(
+            _callData,
+            (address, bytes)
+        );
 
         require(fallbackAddress != address(0), "!invalidFallback");
 
         // transfer to fallback address if forwardFunctionCall fails
-        if (!forwardFunctionCall(_transferId, data, _amount, _asset)) {
+        if (!_prepareAndForward(_transferId, data, _amount, _asset)) {
```

Comment:
this calls a virtual function, how will consumers know this function must *never* fail hard (else funds will be frozen)

Reply:
I would rather try-catch on this, but it's not able to be done in an internal function. Is there any other way to do it?

Reply:
I tried something here.